src/averageAges.js

```diff
@@ -55,7 +71,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const difference = [];
+
+  const children = people.filter(person =>
+    person.mother
+    && (!onlyWithSon || person.sex === 'm'));
+
+  const mothers = people.filter(
+    person => person.sex === 'f'
+      && children.find(child => child.mother === person.name)
+  );
+
+  mothers.forEach(mother => {
+    return children.find(child => {
+      if (child.mother === mother.name) {
+        difference.push(child.born - mother.born);
+      }
+    });
+  });
+
+  return difference.reduce((prev, next) => (
+    prev + next
+  ), 0) / difference.length;
```

Comment:
use reduce instead forEach and separate variable + reduce. Also you do it in one reduce call